pngcanvas.py

```diff
@@ -47,6 +47,16 @@ def gradient_list(start, end, steps):
             for i in range(steps + 1)]
 
 
+def rgb2rgba(rgb):
+    """Take a row of RGB bytes, and convert to a row of RGBA bytes."""
+    rgba = []
```

Comment:
That will return a list of 4-tuples, however the caller is expecting a list. For this to work you'd need something like:

    sum(((rgb[i], rgb[i+1], rgb[i+2], 255) for i in xrange(0, len(rgb), 3)), ())

Which is considerably slower than the current solution. If someone can optimise this further I'd be more than happy to use it, but I tried a few different things, and this seems to be the fastest without dropping to C.